engine/common/rpc/convert/collections.go

```diff
@@ -56,6 +56,34 @@ func MessageToLightCollection(m *entities.Collection) (*flow.LightCollection, er
 	}, nil
 }
 
+func FullCollectionToMessage(c *flow.Collection) ([]*entities.Transaction, error) {
+	if c == nil || c.Transactions == nil {
```

Comment:
is `c.Transactions == nil` needed here? the rest of the logic should still work if transactions are nil.